diesel/src/sqlite/expression/expression_methods.rs

```diff
@@ -0,0 +1,24 @@
+use expression::Expression;
+use super::operators::*;
+
+pub trait SqliteExpressionMethods: Sized {
```

Comment:
To prevent having the "same" method thrice we could do something like this:
```rust
pub trait SqlExpressionMethods: Sized {
    fn collate<T>(self, collate: T) -> Collate<Self, T> where T: CollateKind {
        Collate::new(self)
    }
}

pub trait CollateKind{}
pub mod collate {
    pub struct Binary;
    pub struct NoCase; 
    pub struct RTrim;

    impl CollateKind for Binary {}
    impl CollateKind for NoCase {}
    impl CollateKinde for RTrim {}
}
```

(This is only a suggestion to discuss this with @diesel-rs/contributors )

Reply:
Probably a good idea.
>An application can register additional collating functions using the `sqlite3_create_collation()` interface.

https://sqlite.org/datatype3.html#collating_sequences